ironfish/src/network/peerNetwork.ts

```diff
@@ -1066,6 +1071,11 @@ export class PeerNetwork {
       : (from: BlockHeader) => this.chain.iterateTo(from)
 
     for await (const header of iterationFunction(from)) {
+      if (remainingLookups === 0) {
```

Comment:
I think this technically does an extra lookup as well -- like if remainingLookups is 1, it'll fetch a header, decrement to 0, fetch another header, see that remainingLookups is 0, and break. I don't really mind in this case though because it accomplishes the goal of limiting the returned blocks correctly, so I'd be okay looking at this in a followup